lightkurve/tests/test_periodogram.py

```diff
@@ -330,13 +330,6 @@ def test_error_messages():
     with pytest.raises(ValueError) as err:
         lc.to_periodogram(frequency=np.arange(10), period=np.arange(10))
 
-    # Don't accept NaNs
-    with pytest.raises(ValueError) as err:
-        lc_with_nans = lc.copy()
-        lc_with_nans.flux[0] = np.nan
-        lc_with_nans.to_periodogram()
-    assert('Lightcurve contains NaN values.' in err.value.args[0])
-
```

Comment:
Complementary to the previous comment, I don't have a good feeling about removing a test for invalid input without substitute. One option would be (with the exception kept in place) to replace the check by
`LombScarglePeriodogram.from_lightcurve(lc_with_nans)`
but this probably needs some further discussion if we are willing to have the two methods `to_periodogram` /  `from_lightcurve` behave differently on NaN input.